[SourceKit] [NFC] Rename DEPENDS to LINK_LIBRARIES throughout SourceKit

harlanhaskins opened the pull request:
Currently, SourceKit's CMake functions all use DEPENDS to specify
libraries the targets will link with. This is confusing as it doesn't
behave the same way that add_swift behaves, and implies that
dependencies are created when there aren't.

This is the "proper fix" for #11226.

harlanhaskins commented:
@swift-ci please smoke test

harlanhaskins commented:
@swift-ci please smoke test

harlanhaskins commented:
@swift-ci please smoke test

harlanhaskins commented:
@swift-ci please smoke test

gottesmm commented:
@llvm-beanz Your thoughts?

harlanhaskins commented:
@swift-ci please smoke test

llvm-beanz commented:
The general convention in LLVM is:

- `LINK_COMPONENTS` for LLVM Components
- `LINK_LIBS` for link dependencies that are not LLVM components
- `DEPENDS` for order dependencies

While I'm completely in favor of renaming `DEPENDS` because its use is confusing. I'd rather rename to `LINK_LIBS` rather than `LINK_LIBRARIES` so that we're consistent with all the other LLVM projects.

harlanhaskins commented:
The rest of Swift uses `LLVM_COMPONENT_DEPENDS` instead of `LINK_COMPONENTS`, and I think it would be nice to explicitly namespace these parameters to `LLVM_LINK_COMPONENTS` within the Swift project.

I'd rather this patch not touch the rest of the Swift CMake because it's a very large surface area and I wanted to keep this in line with the existing conventions in the rest of Swift. This is something that I think should be standardized across the projects though.

That said, I'm more than happy to rename `LINK_LIBRARIES` to `LINK_LIBS` for this patch 👍

Thanks for taking a look!

harlanhaskins commented:
@swift-ci please smoke test

harlanhaskins commented:
@llvm-beanz Good to merge once CI passes?

gottesmm commented:
@harlanhaskins Can you rebase the branch before you merge?

harlanhaskins commented:
@gottesmm I wish GitHub would rebase for its web conflicts resolver 😂 (I'm rebasing now)

gottesmm commented:
@harlanhaskins I agree.

harlanhaskins commented:
I did a bad find&replace so I need to manually fix this. Give me just a bit.

harlanhaskins committed:
Rename DEPENDS to LINK_LIBS throughout SourceKit

Currently, SourceKit's CMake functions all use DEPENDS to specify
libraries the targets will link with. This is confusing as it doesn't
behave the same way that add_swift behaves, and implies that
dependencies are created when there aren't.

harlanhaskins commented:
@swift-ci please smoke test

harlanhaskins commented:
@swift-ci please test

harlanhaskins commented:
@swift-ci please smoke test

harlanhaskins commented:
:boat:
